dinosure_course_gordon_fleming/code/06-scheduled-functions.js

```diff
@@ -1 +1,37 @@
 // 06-scheduled-functions
+/**
+ * Executed on the schedule defined in `.root-config.json`.
+ * @param {object} params
+ * @param {PlatformPolicy} params.policy The policy for which the scheduled function is running.
+ * @param {PlatformPolicyholder} params.policyholder The policyholder linked to the policy
+ * @return {ProductModuleAction[] | void} The actions to be queued by the platform.
+ */
+const applyAnnualIncrease = ({ policy, policyholder }) => {
+  // Check if the policy is older than a year
+  const policyStartDate = moment(policy.start_date);
+  const today = policy.module.today ?? moment();
+  const policyAge = today.diff(policyStartDate, 'years');
+
+  // Check for older than a year and 1st of January
+  if (policyAge >= 1 && today.month() === 0 && today.date() === 1) {
+    // Increase the cover by R10 000
+    const newCover = policy.sum_assured + 10000 * 100;
+
+    // Update the cover amount to have the new premium calculated
+    policy.sum_assured = newCover;
+    policy.module.cover_amount = newCover;
```

Comment:
What is the point of assigning this?

Reply:
I see now that the action 'update policy' is doing this and its not needed.

Reply:
I see the `policy.sum_assured = newCover` is not needed, but the `policy.module.cover_amount = newCover` is needed to calculate the new premium with the updated cover amount (at least for how my function is structured).

Reply:
Similar to the above, I can't find where `policy.module.cover_amount` is assigned. If it is, it will be redundant because `policy.sum_assured` should already store that data. You can directly use `newCover` instead. 

```javascript
const newPremium = corePremiumCalc( { cover_amount: newCover, ...policy.module} );
```

Reply:
Here in line 22 the property is assigned:
`policy.module.cover_amount = newCover;`

Isn't the way I already have it then just a different way of re-assigning the cover_amount?

Reply:
The cover_amount would need to be swapped around with the spread of the policy module